### PR TITLE
Feature/delete user by id user

### PR DESCRIPTION
**Pull request by isaidou**
# Pull Request

## Type of Change
- [x] Feature
- [ ] Fix
- [ ] Refactor
- [ ] Documentation Update

---

### Summary:
This Pull Request introduces the **`DELETE /api/users/:id`** endpoint, enabling the deletion of a user by their ID. The implementation follows RESTful API standards and ensures proper error handling for cases where the user does not exist.

### Changes:
- Added the `deleteUserById` method in `UserService` for handling user deletions.
- Implemented `deleteUserById` logic in `UserController` for processing requests and sending consistent API responses.
- Registered a new route **`DELETE /api/users/:id`** in `UserRoutes`.
- Updated Swagger documentation to include the new endpoint.

---

## Motivation
This feature is critical for providing administrators with the ability to manage user accounts effectively by removing inactive or unauthorized users from the system. It adheres to best practices for API design by incorporating robust error handling and clean code principles.

---

### Tests Conducted:
- [x] Verified the **`DELETE /api/users/:id`** endpoint with valid user IDs using Postman.
- [x] Tested edge cases such as non-existent or invalid user IDs to confirm proper error handling.
- [x] Inspected the database to ensure the user was successfully removed when the operation was valid.
- [ ] Pending automated unit tests for `UserService.deleteUserById` and `UserController.deleteUserById`.

---

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have added comments to my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by johnnyhelloworld — APPROVED** [review]

File: src/apps/auth/api/user.controller.ts
Comment (FAEZ10) [inline src/apps/auth/api/user.controller.ts:1]: Faut la classe BadRequest pour mieux gérer les erreurs ou ou utiliser validateDataExists dans BaseService qui le fait directement tu peux regarder product pour t'inspirer

File: src/apps/auth/api/user.route.ts
Comment (FAEZ10) [inline src/apps/auth/api/user.route.ts:1]: Faut utiliser une instance de router pour éviter les conflits entre les routes

File: src/apps/auth/domain/user.service.ts
Comment (FAEZ10) [inline src/apps/auth/domain/user.service.ts:1]: Ça rejoint mon premier commentaire la gestion des erreur n'est pas ouf renvoyer null n'est une bonne chose car tu ajouter beaucoup de gestion d'erreur dans ton controller

**Review by FAEZ10 — COMMENTED** [review]
Y'a des amélioration à faire inspire toi de product car on a passé beaucoup de temps pour faire un truc auquel on peut s'inspirer et avoir un truc un peu commun

File: src/apps/auth/api/user.controller.ts
Comment (isaidou) [reply src/apps/auth/api/user.controller.ts:1]: Ok

File: src/apps/auth/api/user.route.ts
Comment (isaidou) [reply src/apps/auth/api/user.route.ts:1]: ok

File: src/apps/auth/domain/user.service.ts
Comment (isaidou) [reply src/apps/auth/domain/user.service.ts:1]: Envoies les videos de balta-z je vais tester un truc au toilette

**Review by DaniOmer — APPROVED** [review]

**Pull request merged**